### PR TITLE
Add test for `torch.distributed`

### PR DESCRIPTION
**Pull request by rsarm**

### CODE REVIEW

File: checks/apps/deeplearning/pytorch/check_pytorch_dist.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+class pytorch_distr_cnn_base(rfm.RunOnlyRegressionTest):
+    valid_systems = ['lumi:gpu']
+    valid_prog_environs = ['builtin']
+    sourcesdir = 'src'
+    num_tasks = 32
+    num_tasks_per_node = 8
+    num_gpus_per_node = 8
+    variables = {
+        'NCCL_DEBUG': 'INFO',
+        'NCCL_SOCKET_IFNAME': 'hsn0',
```
Comment (mszpindler) [inline checks/apps/deeplearning/pytorch/check_pytorch_dist.py:15]: I assume there are 8 gpus per node in use, shouldn't be also all 4 nics (hsn0...3) involved?

File: checks/apps/deeplearning/pytorch/src/cnn_distr.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+num_epochs = 5
+batch_size_per_gpu = 512
+num_iters = 25
+model_name = 'resnet152'
+
+distr_env = DistributedEnviron()
+dist.init_process_group(backend="nccl")
+world_size = dist.get_world_size()
+rank = dist.get_rank()
+device = 0  # distr_env.local_rank  # since CUDA_VISIBLE_DEVICES=$SLURM_LOCALID
```
Comment (mszpindler) [inline checks/apps/deeplearning/pytorch/src/cnn_distr.py:25]: Is this binding defined somewhere else?

File: checks/apps/deeplearning/pytorch/src/cnn_distr.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    rank=rank,
+    shuffle=False,
+    seed=42
+)
+train_loader = DataLoader(
+    train_set,
+    batch_size=batch_size_per_gpu,
+    shuffle=False,
+    sampler=train_sampler,
+    num_workers=32
```
Comment (mszpindler) [inline checks/apps/deeplearning/pytorch/src/cnn_distr.py:58]: Could we read `num_workers` from the environment?

File: checks/apps/deeplearning/pytorch/src/cnn_distr.py (hunk 2 above)
Comment (rsarm) [reply checks/apps/deeplearning/pytorch/src/cnn_distr.py:25]: Indeed. I made some changes last week from tests that I'm running somewhere else. I still have to try them on lumi. I'm making this PR a draft until I test it.

File: checks/apps/deeplearning/pytorch/src/cnn_distr.py (hunk 3 above)
Comment (rsarm) [reply checks/apps/deeplearning/pytorch/src/cnn_distr.py:58]: yes, I will change that after testing.

**Comment by rsarm** [comment]
Hi @mszpindler Just tested this, it's ready. Could you have a look?

**Review by mszpindler — APPROVED** [review]

**Pull request merged**